docs/topics/Voice_Connections.md

```diff
@@ -2,88 +2,167 @@
 
 Voice connections operate in a similar fashion to the [Gateway](#DOCS_GATEWAY/gateways) connection, however they operate on a different set of payloads, and utilize a separate UDP-based connection for voice data transmission. Because UDP is utilized for both the receiving and transmitting of voice, your client _must_ be able to receive UDP packets, even through a firewall or NAT (see [UDP Hole Punching](https://en.wikipedia.org/wiki/UDP_hole_punching) for more information). The Discord Voice servers implement functionality (see [IP Discovery](#DOCS_VOICE_CONNECTIONS/ip-discovery)) for discovering the local machines remote UDP IP/Port, which can assist in some network configurations.
 
-## Voice Payloads and Events
+## Voice Events
 
-### Voice Data Packet
-
-The voice data packet is a payload with an [encrypted voice packet header](#DOCS_VOICE_CONNECTIONS/encrypted-voice-packet-header-structure). The rest of the bytes in the payload are encrypted Opus audio data.
-
-###### Voice Events
+###### Voice OP Codes
 
 | Code | Name | Description |
 |--------|----------|-----------------|
-| 0 | Identify | used to begin a voice websocket connection |
-| 1 | Select Protocol | used to select the voice protocol |
-| 2 | Ready | used to complete the websocket handshake |
-| 3 | Heartbeat | used to keep the websocket connection alive |
-| 4 | Session Description | used to describe the session |
-| 5 | Speaking | used to indicate which users are speaking |
+| 0 | Identify | begin a voice websocket connection |
+| 1 | Select Protocol | select the voice protocol |
+| 2 | Ready | complete the websocket handshake |
+| 3 | Heartbeat | keep the websocket connection alive |
+| 4 | Session Description | describe the session |
+| 5 | Speaking | indicate which users are speaking |
+| 6 | Heartbeat ACK | sent immediately following a received client heartbeat |
+| 7 | Resume | resume a connection |
+| 8 | Heartbeat Interval | the continuous interval in milliseconds after which the client should send a heartbeat |
+| 9 | Resumed | acknowledge Resume |
+| 13 | Client Disconnect | another client has disconnected |
+
+###### Voice Close Event Codes
```

Comment:
Correct table for this;

| Code | Name |
|---------|---------|
| 4016 | Unknown Encryption Mode |
| 4015 | Voice server crashed |
| 4014 | Disconnected | 
| 4012 | Unknown Protocol |
| 4011 | Server not found |
| 4009 | Session timeout |
| 4006 | Session no longer valid |
| 4005 | Already authenticated | 
| 4004 | Authentication failed |
| 4003 | Not authenticated |
| 4001 | Unknown opcode |